api/app/src/v2/Service/CourtOrderService.php

```diff
@@ -18,11 +21,121 @@ class CourtOrderService
     public function __construct(
         private readonly CourtOrderRepository $courtOrderRepository,
         private readonly UserRepository $userRepository,
+        private readonly ReportRepository $reportRepository,
+        private readonly ClientRepository $clientRepository,
+        private readonly DeputyRepository $deputyRepository,
         private readonly EntityManagerInterface $entityManager,
         private readonly LoggerInterface $logger,
     ) {
     }
 
+    private function transformDate(?string $dateValue, string $fieldName, string $format): ?string
+    {
+        if (empty($dateValue)) {
+            return null;
+        }
+
+        try {
+            $date = new \DateTimeImmutable($dateValue);
+            return $date->format($format);
+        } catch (\Exception $e) {
+            // Optional: log the error for debugging
+            error_log(sprintf(
+                "Date transform failed for %s: %s (value: %s)\n",
+                $fieldName,
+                $e->getMessage(),
+                $dateValue
+            ));
+            return null;
+        }
+    }
+
+    public function transformReportDates(array $reportArray): ?array
```

Comment:
If we're setting this as a public method, we should be checking these keys exist in the provided array.

Reply:
I've set to private though :)